plugins/Denizen/scripts/baguette.dsc

```diff
@@ -0,0 +1,39 @@
+baguette:
+    type: item
+    display name: <&f>Baguette
+    material: bread
+    flags:
+        bites: 4
+        allergy: true
+        # Allergy flag is for future project
+        uuid: <util.random_uuid>
+    lore:
+    - <&7>It's so long that it
+    - <&7>takes 5 bites to eat.
+    - <&7>
+    - <&7>Bites left: <&8>5
+    recipes:
+        1:
+            type: shapeless
+            input: bread|bread|bread|bread|bread
+
+baguette_eat:
+    type: world
+    debug: false
+    events:
+        on player consumes baguette:
+            - define item <player.item_in_hand>
+            - if <[item].flag[bites]> <= 0:
+                - feed amount:17 saturation:<[item].flag[bites].add[4]>
+                - stop
+            - determine passively cancelled
+            - definemap bite_count:
+                4: 2
+                3: 4
+                2: 6
+                1: 8
+            - define amount <[bite_count].get[<[item].flag[bites]>]>
+            - feed amount:<[amount]> saturation:<[item].flag[bites].add[4]>
+            - inventory flag slot:hand bites:--
+            - inventory adjust slot:hand "lore:<[item].lore.get[1].to[-2].separated_by[<n>]><n><&7>Bites left: <&8><[item].flag[bites]>"
```

Comment:
this lore change seems really hacky? It looks like you're just trying to change line 4? If so, just use <[item].lore.set[(your stuff)].at[4]>

Reply:
funky's solution is equally bad, but this wasn't it. In no case do you need to be rewriting the lore in it's entirety. Both of you seem to be forgetting that lore is a list and can be edited per-item